src/Icons/src/Registry/LocalSvgIconRegistry.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Icons\Registry;
+
+use Symfony\Component\DomCrawler\Crawler;
+use Symfony\Component\Finder\Finder;
+use Symfony\UX\Icons\Exception\IconNotFoundException;
+use Symfony\UX\Icons\IconRegistryInterface;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @internal
+ */
+final class LocalSvgIconRegistry implements IconRegistryInterface
+{
+    public function __construct(private string $iconDir)
+    {
+    }
+
+    public function get(string $name): array
+    {
+        if (!file_exists($filename = sprintf('%s/%s.svg', $this->iconDir, str_replace(':', '/', $name)))) {
+            throw new IconNotFoundException(sprintf('The icon "%s" (%s) does not exist.', $name, $filename));
+        }
+
+        $svg = file_get_contents($filename) ?: throw new \RuntimeException(sprintf('The icon file "%s" could not be read.', $filename));
+        $crawler = (new Crawler($svg))->filter('svg');
+        $node = $crawler->getNode(0) ?? throw new \RuntimeException(sprintf('The icon file "%s" is not a valid SVG.', $filename));
+        $attributes = [];
+
+        if ($viewBox = $node->attributes?->getNamedItem('viewbox')?->nodeValue) {
+            $attributes['viewBox'] = $viewBox;
+        }
+
+        return [$crawler->html(), $attributes];
```

Comment:
Can anyone with php-dom experience help me find the equivalent code that doesn't require `symfony/dom-crawer` & `symfony/css-selector`?

Reply:
Minor exception and checks, that is the idea:

```
$doc = new \DOMDocument();
$doc->loadXML($svg);
$svgTag = $doc->firstElementChild;
$html = $doc->saveHTML($svgTag);
$attributes = array_map(fn(\DOMAttr $a) => $a->value, [...$svgTag->attributes]);
```

There may be quicker

Reply:
> `$html = $doc->saveHTML($svgTag);`

This gives me the "outer html" (the entire svg tag), not the guts.

Reply:
Solved but needs some tests.